components/odh-notebook-controller/controllers/notebook_controller_test.go

```diff
@@ -1014,8 +1015,132 @@ var _ = Describe("The Openshift Notebook controller", func() {
+			// cannot create the OAuthClient object
+			time.Sleep(3 * time.Second)
+		})
+
+		It("Should have finalizer set by controller or webhook", func() {
+			By("Waiting for controller to add the finalizer")
+			Eventually(func() []string {
+				var updated nbv1.Notebook
+				err := cli.Get(ctx, notebookKey, &updated)
+				if err != nil {
+					return nil
+				}
+				return updated.Finalizers
+			}, duration, interval).Should(ContainElement(Finalizer),
+				"Expected finalizer '%s' to be added by controller", Finalizer)
+		})
+
+		It("Should create OAuthClient for the Notebook CR", func() {
+			By("Ensuring finalizer exists first")
+			Eventually(func() []string {
+				var updated nbv1.Notebook
+				err := cli.Get(ctx, notebookKey, &updated)
+				if err != nil {
+					return nil
+				}
+				return updated.Finalizers
+			}, duration, interval).Should(ContainElement(Finalizer))
+
+			By("Waiting for controller to create OAuthClient")
+			oauthClientKey := types.NamespacedName{Name: Name + "-" + Namespace + "-oauth-client"}
+			createdOAuthClient := &oauthv1.OAuthClient{}
+			Eventually(func() error {
+				return cli.Get(ctx, oauthClientKey, createdOAuthClient)
+			}, time.Second*90, time.Second*3).Should(Succeed(),
+				"Expected OAuthClient to be created by odh-notebook-controller")
+
+			By("Verifying OAuthClient has proper labels")
+			Expect(createdOAuthClient.Labels).Should(
+				HaveKeyWithValue("notebook-owner", Name),
+				"OAuthClient should have labels linking it to the notebook")
+		})
 
+		It("Should delete OAuthClient when Notebook is deleted", func() {
+			By("Ensuring prerequisite resources exist")
+			Eventually(func() []string {
+				var updated nbv1.Notebook
+				err := cli.Get(ctx, notebookKey, &updated)
+				if err != nil {
+					return nil
+				}
+				return updated.Finalizers
+			}, duration, interval).Should(ContainElement(Finalizer))
+
+			oauthClientKey := types.NamespacedName{Name: Name + "-" + Namespace + "-oauth-client"}
+			Eventually(func() error {
+				return cli.Get(ctx, oauthClientKey, &oauthv1.OAuthClient{})
+			}, duration, interval).Should(Succeed())
+
+			By("Deleting the notebook")
+			var notebookToDelete nbv1.Notebook
+			Expect(cli.Get(ctx, notebookKey, &notebookToDelete)).Should(Succeed())
+			Expect(cli.Delete(ctx, &notebookToDelete)).Should(Succeed())
+
+			By("Verifying notebook is marked for deletion but persists due to finalizer")
+			Eventually(func() bool {
+				var updated nbv1.Notebook
+				if err := cli.Get(ctx, notebookKey, &updated); err != nil {
+					return false
+				}
+				return updated.DeletionTimestamp != nil
+			}, duration, interval).Should(BeTrue(),
+				"Notebook should be marked for deletion but persist due to finalizer")
+
+			By("Verifying controller deletes the OAuthClient")
+			Eventually(func() error {
+				var oauth oauthv1.OAuthClient
+				return cli.Get(ctx, oauthClientKey, &oauth)
+			}, time.Second*90, time.Second*3).Should(
```

Comment:
@coderabbitai is there some constant for the magic numbers? also, 90 seconds seems surprisingly long

Reply:
I believe these numbers came from tests that I was executing previously, where the logic of my implementation wasn't working as expected, sometimes even taking longer times to see the finalizers get added to the resources, deleting the resources properly, etc, but I believe that in its current state, yes, these numbers can be much lower and based on constants for the consistency, no problem on changing that.